### ISSUE TITLE
ExecuteOn with a fixed thread pool doesn't increase concurrency beyond 1

### ISSUE THREAD

**Issue by mattwelke** [micronaut-projects/micronaut-core#6155]
### Expected Behavior

I expected that my server would be able to process 2 requests in parallel.

### Actual Behaviour

The server was only able to process one request at a time. I had a `Thread.sleep(5000);` line in my controller method and invoked the URL twice at about the same time in my web browser. It took about 5 seconds for the first response to come back and about 10 seconds for the second response to come back.

### Steps To Reproduce

1. Use Micronaut Launch to create a new app with every setting at default except changing Java 11 to Java 16.
2. Open the project in an IDE edit the source code.
3. Follow the latest 3.0.1 Micronaut documentation under "Reactive HTTP Request Processing" (https://docs.micronaut.io/latest/guide/).
4.  Edit the `application.yml` file according to the documentation to set the IO thread pool to a fixed pool of size 75. The finished config file looks like:
```yaml
micronaut:
  application:
    name: micronautTest
  executors:
    io:
      type: fixed
      nThreads: 75
``` 
5. Continuing to follow the documentation, annotate your controller method with `@ExecuteOn(ExecutorsService.IO)`. The finished controller looks like:
```java
package com.mattwelke;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.scheduling.TaskExecutors;
import io.micronaut.scheduling.annotation.ExecuteOn;

@Controller("/")
@ExecuteOn(TaskExecutors.IO)
public class UsersController {
    @Get("/{id}")
    String getUser(String id) {
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            e.printStackTrace();
            return "could not get user: " + e.getMessage();
        }
        return "User with ID: " + id;
    }
}

```
6. Run with `./gradlew run`.
7. Open two web browser tabs and load `http://localhost:8080/abc` in each at the same time.
8. Observe that it takes about 10 seconds for the second tab to get its response, inferring that the server was only able to process one  request at a time.

### Environment Information

- Windows 10 Pro
- Oracle OpenJDK 16

### Example Application

https://github.com/mattwelke/micronaut-bug-report-sep-13

### Version

3.0.1

**Comment by graemerocher** [comment]
I cannot reproduce your observations personally, two requests in different tabs run in parallel each taking 5 seconds

I modified the code slightly to do this:

```java
      System.err.println("Processing");
        if (!id.equalsIgnoreCase("2nd")) {
            try {
                Thread.sleep(5000);
            } catch (InterruptedException e) {
                e.printStackTrace();
                return "could not get user: " + e.getMessage();
            }

        }
```

Each each time "Processing" gets printed to the console immediately and when I hit http://localhost:8080/2nd the request is processed immediately.

**Comment by jameskleeh** [comment]
@mattwelke Are you observing this in an environment with limited CPUs?

**Comment by mattwelke** [comment]
> @mattwelke Are you observing this in an environment with limited CPUs?

No, it's a Ryzen desktop processor.

Thanks for taking a look. I eventually figured out that the problem had nothing to do with Micronaut and instead had to do with the client, in this case, the web browser. I ran into an issue described in https://stackoverflow.com/questions/45190458/why-is-my-simple-jetty-server-handling-only-1-request-at-a-time where sometimes, when you use one web browser, you're limited to one connection open at a time to a particular host. So in this case, even though I had two tabs open in Chrome, each pointing to `http://localhost:8080/abc`, only one tab was allowed by Chrome to load at a time.

When I switched to a different approach where I used Chrome and Edge at the same time, each with a tab open to `http://localhost:8080/abc`, both loaded at the same time. I suppose that if I had used a tool like curl instead of a web browser to test from the beginning, I would not have experienced this problem at all.

I followed the links in the Stack Overflow answer to try to understand why web browsers sometimes do this. It seems for you, your web browser didn't do this. But unfortunately, I wasn't able to find more information about this. But at least I now understand that the `ExecuteOn` annotation works as intended. Indeed, I was able to tweak the number of threads in the IO thread pool and get exactly the behavior I expected each time I tweaked it (e.g. seeing two browsers processed in 5 seconds and a third browser processed in 10 seconds when threads set to 2, inferring a concurrency of 2). I also had success testing out setting up a new thread pool with a custom name and using `ExecuteOn` to execute the method on that new thread pool.
